src/extension.ts

```diff
@@ -199,6 +200,23 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
     ext.managedIdentityBranchDataProvider = new ManagedIdentityBranchDataProvider();
     ext.v2.api.resources.registerAzureResourceBranchDataProvider(AzExtResourceType.ManagedIdentityUserAssignedIdentities, ext.managedIdentityBranchDataProvider);
 
+
+
+    // Register the exportAuthRecord callback for session changes
+    registerEvent(
+        'treeView.onDidChangeSessions',
+        vscode.authentication.onDidChangeSessions,
+        exportAuthRecord
+    );
+
+    // Also call exportAuthRecord once on activation to ensure the auth record is exported at least once
+    // (onDidChangeSessions is not guaranteed to fire on activation)
+    void exportAuthRecord({
```

Comment:
Instead of setting up a fake `IActionContext`, you can use `callWithTelemetryAndErrorHandling` from the `@microsoft/vscode-azext-utils` package. It's probably OK to use the exact same event ID but you could do a different one.

As alluded to in the other comment above, I'd still `void` this, rather than `await`ing it.